pkg/controller/control.go

```diff
@@ -825,6 +825,35 @@ func (c *Controller) ReadAt(b []byte, off int64) (int, error) {
 	return n, err
 }
 
+func (c *Controller) UnmapAt(length uint32, off int64) (int, error) {
+	// TODO: Need to fail unmap requests
+	//  if the volume is purging snapshots or creating backups.
+	c.RLock()
```

Comment:
For the backup creation, if some removed files are unloaded to the remote target before the trim reclaiming it, the backup would contain some garbage data. And the checksum of a volume restored from this backup will be different from the original volume (after the trim). But I think this difference is acceptable.

For the purge, some blocks of the removed files may be retained. I am not sure if it's possible.

In general, it would not crash the data hence I didn't resolve this issue in the 1st version of the trim feature.